### PR TITLE
Report warnings for COM interfaces with instance properties or events

### LINKED ISSUE #87158: GeneratedComInterface interfaces should emit a diagnostic for non-static properties and events
Properties are not a concept in COM, and .NET has historically provided an OLE-based interop solution through the `IConnectionPoint` infrastructure. We aren't supporting the `IConnectionPoint`-based events in the COM source generator, so we should issue a diagnostic for events and properties. We can either issue a warning that no interop code will be generated, or we can issue an error blocking the scenario (as there's not a particularly good reason to allow it).

If we so desire, we could allow DIM-implemented properties/events as they would not require the generator to emit an implementation, but I think we could wait on that until we get sufficient user feedback about it (and even then, we could point users to the "roles" feature if C# has added it by then).

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Properties and events are not concepts in COM, and the limited cases where the concept translates over, it's limited to IDispatch-based scenarios.

To keep ourselves from being pushed into a corner in design, we will issue warnings for these types of members on `GeneratedComInterface` interfaces and not generate code for them.

Fixes #87158

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs [hunk 1]
```diff
@@ -195,6 +197,32 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
...
+                {
+                    SymbolKind.Property => GeneratorDiagnostics.InstancePropertyDeclaredInInterface,
+                    SymbolKind.Event => GeneratorDiagnostics.InstanceEventDeclaredInInterface,
+                    _ => null
+                };
+
+                if (descriptor is not null)
+                {
+                    diagnostics.Add(member.CreateDiagnosticInfo(descriptor, member.Name, symbol.ToDisplayString()));
+                }
```
Comment (jtschuster) [inline src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs:220]: Could we extract this into a method in ComMethodInfo.cs and call it where we loop over the members in `GetMethodFromInterface` (and we could maybe rename that method).

File: src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/Resources/Strings.resx [hunk 2]
```diff
@@ -214,7 +214,7 @@
     <value>All methods must be declared in the same partial definition of a 'GeneratedComInterface'-attributed interface type to ensure reliable calculation for virtual method table offsets.</value>
   </data>
   <data name="MethodNotDeclaredInAttributedInterfaceMessage" xml:space="preserve">
-    <value>The method '{0}' is declared on a different partial definition of the interface '{1}' than the definition that has the 'GeneratedComInterface' attribute</value>
+    <value>The method '{0}' is declared on a different partial definition of the interface '{1}' than the definition that has the 'GeneratedComInterface' attribute. Interop code for '{0}' will not be generated</value>
```
Comment (jtschuster) [inline src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/Resources/Strings.resx:217]: If this is always going to be an error that breaks the build I don't think we need the extra part. I think we should leave the diagnostic as is or make the `MethodNotDeclaredInAttributedInterface` diagnostic a warning.

**Review by jtschuster — COMMENTED** [review]
I think we should also test this with events or properties outside of the GeneratedComInterface definition to make sure it doesn't double warn. 
Also, generally, should we error if we find any members that aren't supported COM methods in the attributed definition, and make user have a separate interface if they want both COM and non-COM members? Right now, a static method in the middle of the COM methods will not warn and won't be generated.

**Comment by AaronRobinsonMSFT** [comment]
> Properties and events are not concepts in COM

That isn't accurate. Properties can be defined using the `propget`, `propput` and `propputref` attributes and they are not required to be on types derived from `IDispatch`. They are typically only expressed in a "property" way in `IDispatch` scenarios but they are valid in COM.

Events are defined using the `source` keyword and do have a fair bit of boilerplate that is best deferred to v2 or until we get a customer.

Not saying we should support either, but they are COM concepts and supported outside of `IDispatch`. I would be supportive of supporting basic properties (that is, `propget`, `propput`) for v1, but am also okay with it being deferred for now.

**Comment by jkoritzinsky** [comment]
> Also, generally, should we error if we find any members that aren't supported COM methods in the attributed definition, and make user have a separate interface if they want both COM and non-COM members? Right now, a static method in the middle of the COM methods will not warn and won't be generated.

I think allowing static members is fine, but we should disallow any instance members that would contribute to the generated vtable in the built-in COM system so as to avoid bad migration experiences where things break badly because vtable offsets are wrong.

**Review by jtschuster — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Failure is unrelated

**Pull request merged**